transmission_interface/test/differential_transmission_loader_test.cpp

```diff
@@ -0,0 +1,469 @@
+#include "transmission_interface/transmission_loader.hpp"
+
+using testing::SizeIs;
+
+class TransmissionPluginLoader
+{
+public:
+  std::shared_ptr<transmission_interface::TransmissionLoader> create(const std::string & type)
+  {
+    try
+    {
+      return class_loader_.createUniqueInstance(type);
+    }
+    catch (std::exception & ex)
+    {
+      std::cerr << ex.what() << std::endl;
+      return std::shared_ptr<transmission_interface::TransmissionLoader>();
+    }
+  }
+
+private:
+  // must keep it alive because instance destroyers need it
+  pluginlib::ClassLoader<transmission_interface::TransmissionLoader> class_loader_ = {
+    "transmission_interface", "transmission_interface::TransmissionLoader"};
+};
+
+TEST(DifferentialTransmissionLoaderTest, FullSpec)
+{
+  // Parse transmission info
+  std::string urdf_to_test = R"(
+    <?xml version="1.0"?>
+    <robot name="robot" xmlns="http://www.ros.org">
+      <ros2_control name="FullSpec" type="system">
+        <joint name="joint1">
+          <command_interface name="velocity">
+            <param name="min">-0.5</param>
+            <param name="max">0.5</param>
+          </command_interface>
+          <state_interface name="velocity"/>
+        </joint>
+        <joint name="joint2">
+          <command_interface name="position">
+            <param name="min">-1</param>
+            <param name="max">1</param>
+          </command_interface>
+          <state_interface name="position"/>
+        </joint>
+        <transmission name="transmission1">
+          <plugin>transmission_interface/DifferentialTransmission</plugin>
+          <actuator name="joint1_motor" role="actuator1">
+            <mechanical_reduction>50</mechanical_reduction>
+          </actuator>
+          <actuator name="joint2_motor" role="actuator2">
+            <mechanical_reduction>-50</mechanical_reduction>
+          </actuator>
+          <joint name="joint1" role="joint1">
+            <offset>0.5</offset>
+            <mechanical_reduction>2.0</mechanical_reduction>
+          </joint>
+          <joint name="joint2" role="joint2">
+            <offset>-0.5</offset>
+            <mechanical_reduction>-2.0</mechanical_reduction>
+          </joint>
+        </transmission>
+      </ros2_control>
+    </robot>
+    )";
+
+  std::vector<hardware_interface::HardwareInfo> infos =
+    hardware_interface::parse_control_resources_from_urdf(urdf_to_test);
+  ASSERT_THAT(infos[0].transmissions, SizeIs(1));
+
+  // Transmission loader
+  TransmissionPluginLoader loader;
+  std::shared_ptr<transmission_interface::TransmissionLoader> transmission_loader =
+    loader.create(infos[0].transmissions[0].type);
+  ASSERT_TRUE(nullptr != transmission_loader);
+
+  std::shared_ptr<transmission_interface::Transmission> transmission;
+  const hardware_interface::TransmissionInfo & info = infos[0].transmissions[0];
+  transmission = transmission_loader->load(info);
+  ASSERT_TRUE(nullptr != transmission);
```

Comment:
Please check best practices comment.

Reply:
The best practices comment says that either the comparison  ASSERT_TRUE(nullptr != transmission) is redundant, or there is possible null pointer dereference. I don't think there's a case where the transmission_loader would be a nullptr, so should I just leave out these comparisons?